G-Loader

antoinechauvn opened the pull request:
Added G-Loader extension

antoinechauvn committed:
Add files via upload

antoinechauvn committed:
Delete G-Loader directory

antoinechauvn committed:
Create test.py

antoinechauvn committed:
Add files via upload

antoinechauvn committed:
Delete test.py

WiredSpast commented:
The extension.json file has to be a file in this repo, not a file in the zip

WiredSpast commented:
The icon file is also to big, the size limit is 80x80px (it will be resized to 40px in the store, so best to use a 40x40px image)

WiredSpast commented:
![image](https://user-images.githubusercontent.com/74211155/213888994-f2d38125-f3f2-4747-9c25-c34771626216.png)
The files should also be directly in the root folder of the zip not in a subfolder (unless you edit the command to say `G-Loader/__main__.py` instead of `__main__.py`)

WiredSpast commented:
![image](https://user-images.githubusercontent.com/74211155/213889080-7e3eaec4-e51e-4f06-900d-e00584dc8747.png)
The command has to include the cookie and file as well, so it should be
```json
"commands": {
  "default": ["python", "__main__.py", "-c", "{cookie}", "-p", "{port}", "-f", "{filename}"]
},
```

antoinechauvn committed:
Delete extension.zip

antoinechauvn committed:
Delete icon.png

antoinechauvn committed:
Create test

antoinechauvn committed:
Add files via upload

antoinechauvn committed:
Delete test

antoinechauvn commented:
Updated

WiredSpast commented:
You still forgot to add the `extension.json` file into this repo

antoinechauvn committed:
Add files via upload

antoinechauvn commented:
Done

WiredSpast commented on store/extensions/G-Loader/extension.json:
```diff
@@ -0,0 +1,38 @@
+
+    "framework": {
+      "name": "G-Python",
+      "version": "0.1.5"
```
You appear to have used 0.1.6 not 0.1.5
```suggestion
      "version": "0.1.6"

```

WiredSpast commented:
The extension doesn't terminate when G-Earth does, it should
(This happens because the UI thread keeps running while the extension thread dies)

WiredSpast commented:
You're extension seems to have an extra dependency next to G-Python, I recommend using PyExtensionSetup and a `requirements.txt` file, the extension setup will make sure the user has python and the necesarry requirements installed.

The launch command would look like this:
```json
"default": ["java", "-jar", "PyExtensionSetup.jar", "-e", "__main__.py", "-v", "3.2.0", "-c", "{cookie}", "-p", "{port}", "-f", "{filename}"]
```
and the requirements.txt would have this in it:
```
g-python==0.1.6
PySide6==6.4.2
```
and then include both the requirements.txt file and PyExtensionSetup.jar in your extension.zip

PyExtensionSetup release: https://github.com/WiredSpast/PyExtensionSetup/releases/latest

antoinechauvn committed:
Update store/extensions/G-Loader/extension.json

Co-authored-by: WiredSpast <74211155+WiredSpast@users.noreply.github.com>

antoinechauvn committed:
Add files via upload

antoinechauvn committed:
Delete extension.zip

antoinechauvn committed:
Add files via upload

antoinechauvn commented:
Done

WiredSpast commented:
The extension still remains running after G-Earth closes, you can fix this by using a timer from the threading package which checks whether the extension's connection has been closed or not every second:

```py
# Check if extension is still connected to G-Earth every second
def checkExtensionActive():
  if ext.is_closed():
    # Close window and stop if extension isn't connected
    window.close(); # This is pseudo code I haven't looked at how to close a window in the framework you're using
        
    # Repeat every second
    Timer(1.0, checkExtensionActive).start()
    
checkExtensionActive()
```

I've also discussed the permanently open UI with sirjonas and we've come to an agreement that extensions in the store shouldn't have a permanently open UI. It should start hidden and be opened by the green playbutton, and the window should then be able to be closed again (either by use of the same playbutton or by an X button):

```py
# window.toggle is again pseudocode
ext.on_event('double_click', window.show)
```

The extension is also currently always on top, it shouldn't unless you've allowed a user to say that they want it to remain on top (using for a checkbox like G-Earth has in it's extra tab).

antoinechauvn committed:
Delete extension.zip

antoinechauvn committed:
Add files via upload

antoinechauvn commented:
Updated

WiredSpast committed:
JSON formatting fault

antoinechauvn committed:
Delete extension.zip

antoinechauvn committed:
Add files via upload

antoinechauvn committed:
Delete store/extensions/G-Loader/extension.zip

antoinechauvn committed:
Add files via upload

antoinechauvn commented:
- Fixed Qt autostart due to G-Earth running all extensions files.
```python
def display():
    try:
        window = engine.rootObjects()[0]
    except:
        engine.load(qml_file)
    else:
        window.setVisible(not window.isVisible())
```

- Fixed Window size and close button
- Fixed Habbo.com Api maintenance

WiredSpast commented on store/extensions/G-Loader/extension.json:
```diff
@@ -0,0 +1,37 @@
+        "clients": ["Nitro"]
+    },
+
+    "submissionDate": "21-01-2023 21:05:00",
+    "updateDate": "21-01-2023 21:05:00",
```
Might be better to set these to today, otherwise your extension is gonna be low down in the store

WiredSpast commented:
The code in your zip file currently doesn't match to the code in your repo, make sure these match

antoinechauvn committed:
Update extension.json
